src/core/qgsvectorlayerutils.h

```diff
@@ -155,6 +156,17 @@ class CORE_EXPORT QgsVectorLayerUtils
      */
     static QgsFeature duplicateFeature( QgsVectorLayer *layer, const QgsFeature &feature, QgsProject *project, int depth, QgsDuplicateFeatureContext &duplicateFeatureContext SIP_OUT );
 
+    /**
+     * Gets the feature source from a weak QgsVectorLayer pointer.
+     * This method is thread-safe but will block the main thread for execution. Executing it from the main
```

Comment:
This isn't quite true, right? It makes it sound like the main thread is blocked for the duration of the call -- but it's only blocked for the duration of getting the source, right? The rest of the the the main thread is humming away doing whatever else it's up to.

Reply:
if by "the call" you mean "consuming the iterator", then your statement is true. Are you proposing to replace the word "execution" with something else?

Reply:
Yeah, but honestly I can't come up with an elegant way to word this!